pkg/asset/agent/image/oc.go

```diff
@@ -126,19 +138,54 @@ func (r *release) extractFileFromImage(log logrus.FieldLogger, image, file, cach
 		cmd = fmt.Sprintf(templateImageExtract, file, cacheDir, image)
 	}
 
-	log.Debugf("extracting %s to %s, %s", file, cacheDir, cmd)
+	logrus.Debugf("extracting %s to %s, %s", file, cacheDir, cmd)
 	_, err := retry.Do(r.config.MaxTries, r.config.RetryDelay, execute, log, r.executer, pullSecret, cmd)
 	if err != nil {
 		return "", err
 	}
 	// set path
 	path := filepath.Join(cacheDir, path.Base(file))
-	log.Info("Successfully extracted base ISO from the release")
-	log.Debugf("Base ISO %s cached at %s", file, path)
 	return path, nil
 }
 
-func execute(log logrus.FieldLogger, executer executer.Executer, pullSecret string, command string) (string, error) {
+// Check if there is a different base ISO in the release payload
+func (r *release) verifyCacheFile(log logrus.FieldLogger, image, file, pullSecret, architecture string, mirrorConfig []mirror.RegistriesConfig) (bool, error) {
+
+	tempDir, err := ioutil.TempDir("", "cache")
+	if err != nil {
+		return false, err
+	}
+
+	// Get the file containing the coreos sha256 hash
+	shaFilename := fmt.Sprintf(coreOsSha256FileName, architecture)
+	shaFile, err := r.extractFileFromImage(log, image, shaFilename, tempDir, pullSecret, mirrorConfig)
+	if err != nil {
+		return false, err
+	}
+
+	defer os.RemoveAll(tempDir)
+
+	payloadSha256, err := ioutil.ReadFile(shaFile)
+	if err != nil {
+		return false, err
+	}
+
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return false, err
+	}
+	fileHash := sha256.Sum256(data)
+
+	// Compare the hash of cached file to the hash retrieved from payload
+	if hex.EncodeToString(fileHash[:]) == string(payloadSha256) {
+		// cached file is the same as in payload
+		return true, nil
+	}
+	logrus.Debugf("Cached file %s is not most recent", file)
+	return false, err
```

Comment:
`err` is always `nil` here though, right?

Reply:
Yes, changed to make it obvious.